internal/services/securitycenter/security_center_storage_defender_resource.go

```diff
@@ -79,6 +82,12 @@ func (s StorageDefenderResource) Arguments() map[string]*schema.Schema {
 			Optional: true,
 			Default:  false,
 		},
+
+		"scan_results_event_grid_topic_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
```

Comment:
This can be simplified using common schemas
```suggestion
		"scan_results_event_grid_topic_id": commonschema.ResourceIDReferenceOptional(&topics.TopicID{})
```